Remove claim for Discord avatar

martincostello opened the pull request:
Rather than maintain logic to understand how Discord builds up avatar URLs and handles different image formats, this PR removes the computed "avatar URL" claim for the v6 release while we have the opportunity to make breaking changes.

Consumers can instead author their own `MapCustomJson(...)` on the `ClaimActions` and derive the URI based on their own requirements.

As this is pretty easy to do using the extensibility that already exists, I think this is a long-term more preferable and maintainable solution than us trying to keep in step with whatever it is that Discord is doing for their avatar image storage.

If we decide this is the route we want to go down, I'll add an example implementation that mirrors what previous versions did with a tweak to support the animated case into the Markdown document. Users can then copy-paste that into their app configuration if they need it.

We could also optionally add `[Obsolete]` to these properties in the dev branch for v5 to give some advance notice to it being removed in v6.

Relates to #584.

martincostello committed:
Remove claim for Discord avatar

Rather than maintain logic to understand how Discord builds up avatar URLs and handles different image format, remove the computed "avatar URL" claim.
Consumers can instead author their own MapCustomJson on the ClaimActions and derive the location based on their own requirements.

kevinchalet reviewed (approved):
Sounds like the most reasonable approach 👍🏻

martincostello committed:
Add sample code for avatar claim

Add some sample code to the documentation showing how to map the user's avatar URL to a claim.

martincostello committed:
Fix links

Turns out # links don't work in Markdown docs.
